### PR TITLE
Fix Segment Mutation lock

### PR DESCRIPTION
**Pull request by xiaofan-luan**
fix the segment mutation lock in flowgraph
Signed-off-by: xiaofan-luan <xiaofan.luan@zilliz.com>

### CODE REVIEW

File: internal/datanode/channel_meta.go [hunk 1]
```diff
@@ -512,6 +512,7 @@ func (c *ChannelMeta) mergeFlushedSegments(seg *Segment, planID UniqueID, compac
 
 	log.Info("merge flushed segments")
 	c.segMu.Lock()
+	defer c.segMu.Lock()
```
Comment (czs007) [inline internal/datanode/channel_meta.go:515]: typo. 
defer c.segMu.Unlock

Comment (czs007) [inline internal/datanode/channel_meta.go:515]: typo.
defer c.segMu.Unlock()

Comment (xiaofan-luan) [reply internal/datanode/channel_meta.go]: done

**Pull request merged**